pandas/core/window_/aggregators.py

```diff
@@ -0,0 +1,136 @@
+import abc
+from typing import Optional
+
+import numpy as np
+
+from pandas._typing import Scalar
+
+
+class BaseAggregator(abc.ABC):
+    """Interface to return the current value of the rolling aggregation at the current step"""
+    def __init__(self, values: np.ndarray) -> None:
+        self.values = values
+
+    @abc.abstractmethod
+    def query(
+        self, start: int, stop: int, previous_start: int, previous_end: int
+    ) -> Scalar:
+        """
+        Computes the result of an aggregation for values that are between
+        the start and stop indices
+
+        Parameters
+        ----------
+        start : int
+            Current starting index
+
+        stop : int
+            Current stopping index
+
+        previous_start : int
+            Prior starting index
+
+        previous_end : int
+            Prior ending index
+
+        Returns
+        -------
+        Scalar
+            A scalar value that is the result of the aggregation.
+
+        """
+
+
+class AggKernel(abc.ABC):
+    """Interface that computes the aggregation value"""
+    def __init__(self):
+        self.count = 0
+
+    @abc.abstractmethod
+    def finalize(self):
+        """Return the final value of the aggregation."""
```

Comment:
Can you please add `make_aggregator` as an abstract class method here? We forgot to put it here in the spec.